pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -54,6 +54,9 @@ const (
 	gceAffinityTypeClientIP = "CLIENT_IP"
 	// AffinityTypeClientIPProto - affinity based on Client IP and port.
 	gceAffinityTypeClientIPProto = "CLIENT_IP_PROTO"
+
+	operationPollInterval        = 3 * time.Second
+	operationPollTimeoutDuration = 7 * time.Minute
```

Comment:
7 minutes is an arbitrary cap that I set, based on the fact that I've seen some disk attach request take a few minutes to succeed. I'm fine with either lowering or raising it. But, I think the poll time must be capped: if there is a server side issue, for example, where the operation is never set to "DONE" we don't want this to block indefinitely (as the old code would do).

When the timeout is hit, the request fails with error, and it is up to the caller to determine what to do (GCE PD attach/detach will retry).